contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -73,8 +81,25 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 			logger.Errorf("Error during marshaling of the message: %v", err)
 			return err
 		}
-		if err := stanutil.Publish(s.natssConn, ch, &message, logger); err != nil {
+		s.natssConnMux.Lock()
+		currentNatssConn := s.natssConn
+		s.natssConnMux.Unlock()
+		if currentNatssConn == nil {
+			return fmt.Errorf("No Connection to NATSS")
+		}
+		if reflect.ValueOf(currentNatssConn).IsNil() {
+			// Informing SubscriptionsSupervisor to re-establish connection to NATSS.
+			s.connect <- struct{}{}
+			return fmt.Errorf("No Connection to NATSS")
+		}
+		if err := stanutil.Publish(currentNatssConn, ch, &message, logger); err != nil {
 			logger.Errorf("Error during publish: %v", err)
+			if err.Error() == stan.ErrConnectionClosed.Error() {
+				logger.Error("Connection to NATSS has been lost, attempting to reconnect.")
+				// Informing SubscriptionsSupervisor to re-establish connection to NATSS.
+				s.connect <- struct{}{}
```

Comment:
We can be fancy and make this a non-blocking insertion. Something like replace this with `s.signalReconnect()` (not a good name). And then have:

```golang
func (s *SubscriptionsSupervisor) signalReconnect() {
    select {
    case s.connect <- struct{}{}:
      // Sent.
    default:
        // The Channel is already full, so a reconnection attempt will occur.
    }
}
```

If we do this, then all writing to the channel should be via the `s.signalReconnect()` function.

Reply:
ok, sounds good thanks for suggestion.